### PR TITLE
delete pr tag on merge/close of a pr

### PR DESCRIPTION
**Pull request by dbasunag**
When a PR is merged or closed, delete the associated tag from the registry.

### CODE REVIEW

File: webhook_server_container/libs/github_api.py [hunk 1]
```diff
@@ -759,6 +759,43 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
...
+            if rc and pr_tag in output:
+                # delete the tag:
+                rc, _, _ = run_command(
+                    command=f"{base_regctl_command} tag delete {self.container_repository}:{pr_tag}",
+                    log_prefix=self.log_prefix,
+                    verify_stderr=True,
+                )
+                if rc:
+                    self.pull_request.create_issue_comment(f"Successfully removed PR tag: {pr_tag}.")
+            else:
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:792]: This is not neeed.

File: webhook_server_container/libs/github_api.py [hunk 2]
```diff
@@ -759,6 +759,43 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
                 issue.edit(state="closed")
                 break
 
+    @ignore_exceptions(logger=FLASK_APP.logger)
+    def delete_remote_tag_for_merged_or_closed_pr(self, hook_action):
+        pr_tag = f"pr-{self.pull_request.number}"
+        # run regctl as a container:
+        base_regctl_command = (
+            "podman container run -i --rm --net host  -v regctl-conf:/home/appuser/.regctl/ "
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:767]: why do you need this `regctl-conf:/home/appuser/.regctl/` ?
you are not persist anything.

Comment (myakove) [inline webhook_server_container/libs/github_api.py:767]: `podman container run` > `podman run` ?

**Review by myakove — CHANGES_REQUESTED** [review]

File: webhook_server_container/libs/github_api.py (hunk 2 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:767]: I basically copied over the example from the readme file: https://github.com/regclient/regclient/blob/main/docs/install.md#running-as-a-container. But on trying locally without it, it looks like without it the previous authentication was ignored 
```
dbasunag@dbasunag-mac github-webhook-server % podman container run -i --rm --net host ghcr.io/regclient/regctl:latest tag ls quay.io/openshift-cnv/cnv-tests-github --verbosity debug                     
time="2024-06-24T16:23:46Z" level=debug msg="regclient initialized" VCSRef=766ee6291f882778207ff42207f9ca8b1da54e57-dirty VCSTag=v0.6.1
time="2024-06-24T16:23:46Z" level=debug msg="Listing tags" host=quay.io repository=openshift-cnv/cnv-tests-github
time="2024-06-24T16:23:46Z" level=debug msg="http req" method=GET url="https://quay.io/v2/openshift-cnv/cnv-tests-github/tags/list" withAuth=false
time="2024-06-24T16:23:47Z" level=debug msg="Auth request parsed" challenge="[{bearer map[realm:https://quay.io/v2/auth scope:repository:openshift-cnv/cnv-tests-github:pull service:quay.io]}]"
time="2024-06-24T16:23:47Z" level=debug msg="http req" method=GET url="https://quay.io/v2/openshift-cnv/cnv-tests-github/tags/list" withAuth=true
time="2024-06-24T16:23:47Z" level=debug msg="Auth request parsed" challenge="[{bearer map[realm:https://quay.io/v2/auth scope:repository:openshift-cnv/cnv-tests-github:pull service:quay.io]}]"
time="2024-06-24T16:23:47Z" level=debug msg="Failed to handle auth request" Err=unauthorized URL="https://quay.io/v2/openshift-cnv/cnv-tests-github/tags/list"
failed to list tags for quay.io/openshift-cnv/cnv-tests-github:latest: unauthorized
dbasunag@dbasunag-mac github-webhook-server % 
```

Comment (myakove) [reply webhook_server_container/libs/github_api.py:767]: ok

Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:767]: Yes. I will update.

File: webhook_server_container/libs/github_api.py (hunk 1 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:792]: Ok, removing it.

File: webhook_server_container/libs/github_api.py [hunk 3]
```diff
@@ -759,6 +759,42 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
...
 
+    @ignore_exceptions(logger=FLASK_APP.logger)
+    def delete_remote_tag_for_merged_or_closed_pr(self, hook_action):
+        pr_tag = f"pr-{self.pull_request.number}"
+        # run regctl as a container:
+        base_regctl_command = (
+            "podman run -i --rm --net host  -v regctl-conf:/home/appuser/.regctl/ ghcr.io/regclient/regctl:latest"
+        )
+        registry_info = self.container_repository.split("/")
+        registry_url = "" if len(registry_info) < 2 else registry_info[0]
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:770]: sorry, should be 3

quay.io/openshift-cnv/cnv-tests-github << with domain
myakove/my-image >> without domain

**Review by myakove — CHANGES_REQUESTED** [review]

File: webhook_server_container/libs/github_api.py [hunk 4]
```diff
@@ -759,6 +759,42 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
...
+    @ignore_exceptions(logger=FLASK_APP.logger)
+    def delete_remote_tag_for_merged_or_closed_pr(self, hook_action):
+        pr_tag = f"pr-{self.pull_request.number}"
+        # run regctl as a container:
+        base_regctl_command = (
+            "podman run -i --rm --net host  -v regctl-conf:/home/appuser/.regctl/ ghcr.io/regclient/regctl:latest"
+        )
+        registry_info = self.container_repository.split("/")
+        registry_url = "" if len(registry_info) < 3 else registry_info[0]
+        self.app.logger.info(f"Using registry url: {registry_url}")
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:771]: This is not needed and also can be empty
we print the command already

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — CHANGES_REQUESTED** [review]

File: webhook_server_container/libs/github_api.py (hunk 4 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:771]: True. Will remove

File: webhook_server_container/libs/github_api.py [hunk 5]
```diff
@@ -759,6 +759,42 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
                 issue.edit(state="closed")
                 break
 
+    @ignore_exceptions(logger=FLASK_APP.logger)
+    def delete_remote_tag_for_merged_or_closed_pr(self, hook_action):
+        pr_tag = f"pr-{self.pull_request.number}"
+        # run regctl as a container:
+        base_regctl_command = (
+            "podman run -i --rm --net host  -v regctl-conf:/home/appuser/.regctl/ ghcr.io/regclient/regctl:latest"
```
[the remark this answers is not in the document]
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:767]: We don't need `-i`. Removing

**Review by myakove — APPROVED** [review]

File: webhook_server_container/libs/github_api.py [hunk 6]
```diff
@@ -759,6 +759,41 @@ def close_issue_for_merged_or_closed_pr(self, hook_action):
...
+        base_regctl_command = (
+            "podman run --rm --net host  -v regctl-conf:/home/appuser/.regctl/ ghcr.io/regclient/regctl:latest"
+        )
+        registry_info = self.container_repository.split("/")
+        registry_url = "" if len(registry_info) < 3 else registry_info[0]
+        # First we need to execute regctl login command before we can delete the tag:
+        rc, _, _ = run_command(
+            command=f"{base_regctl_command} registry login {registry_url} -u {self.container_repository_username} "
+            f"-p {self.container_repository_password}",
+            verify_stderr=True,
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:775]: Do not `verify_stderr` since the command have warnings when running it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: webhook_server_container/libs/github_api.py (hunk 6 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:775]: ok.

**Pull request merged**